internal/server/statvar_group.go

```diff
@@ -187,18 +187,49 @@ func (s *Server) GetStatVarGroupNode(
 						DisplayName:       t.SubjectName,
 						SpecializedEntity: computeSpecializedEntity(svg, t.SubjectID),
 					})
+				allIDs = append(allIDs, t.SubjectID)
 			} else if t.Predicate == "memberOf" {
 				// Children SV
 				result.ChildStatVars = append(result.ChildStatVars,
 					&pb.StatVarGroupNode_ChildSV{
 						Id: t.SubjectID,
 					})
-				childStatVars = append(childStatVars, t.SubjectID)
+				allIDs = append(allIDs, t.SubjectID)
 			}
 		}
 	}
-	// TODO(shifucun): Filter sv and svg given place information
-	log.Println(places)
+
+	// Check if stat data exists for given places
+	statExistence, err := checkStatExistence(ctx, s.store, allIDs, places)
+	if err != nil {
+		return nil, err
+	}
+	// Filter parent stat var groups
+	filteredParentStatVarGroups := []string{}
+	for _, item := range result.ParentStatVarGroups {
+		if c, ok := statExistence[item]; ok && c == len(places) {
+			filteredParentStatVarGroups = append(filteredParentStatVarGroups, item)
+		}
+	}
```

Comment:
Can this be a function perhaps? (maybe have TODO)

Reply:
Added a to do for a generic way of filtering.